src/bindings/python/src/pyopenvino/graph/model.cpp

```diff
@@ -701,9 +706,127 @@ void regclass_graph_Model(py::module m) {
         return "<" + class_name + ": '" + self.get_friendly_name() + "'\ninputs[\n" + inputs_str + "\n]\noutputs[\n" +
                outputs_str + "\n]>";
     });
+    model.def("get_rt_info",
+              (PyRTMap & (ov::Model::*)()) & ov::Model::get_rt_info,
+              py::return_value_policy::reference_internal,
+              R"(
+                Returns PyRTMap which is a dictionary of user defined runtime info.
+
+                :return: A dictionary of user defined data.
+                :rtype: openvino.runtime.RTMap
+             )");
+    model.def(
+        "get_rt_info",
+        [](const ov::Model& self, const py::list& path) -> py::object {
+            std::vector<std::string> cpp_args;
+            for (size_t i = 0; i < path.size(); i++) {
+                cpp_args.emplace_back(path[i].cast<std::string>());
+            }
+            return Common::utils::from_ov_any(self.get_rt_info<ov::Any>(cpp_args));
+        },
+        py::arg("path") = py::list(),
```

Comment:
As I think of it right now, we have `(PyRTMap & (ov::Model::*)()) & ov::Model::get_rt_info` overload which translates to `obj.get_rt_info()`. Now we want to introduce `[](const ov::Model& self, const py::list& path) -> py::object` which is roughly translated to `obj.get_rt_info(path=[])` and there is another one `obj.get_rt_info(path="")`... these are virtually the same (when no argument is passed) and rely on internal resolution order.

I think this is bad, as potentially small reorders in this file will result in taking either overload before others.

Let's remove all default arguments in `get/set/has_rt_info` overloads. First, this will remove a heavy dependency on resolution order to keep compatible. Second, it will force users to pass `path` either in form of a string or list -- and this should a design, right? I think there is no case where runtime attribute key is just empty string... at least I would be very surprised if that works.

(*) - mark for related comments